JW-1344: Configs for zuul server daemons

jacek-izykowski opened the pull request.

Commit:
Zuul and Apache2 config files for winci-zuulv2

Commit:
Zuul configuration cleanup.

krdln committed:
reorganize config files into directories

kaweue commented on zuul/apache2.conf:
```diff
@@ -0,0 +1,231 @@
+# see README.Debian for details.
+
+# Include generic snippets of statements
+IncludeOptional conf-enabled/*.conf
```
We should not modify generic apache.config file. Our config file may be added here mods-enabled/ or conf-enabled/ or sites-enabled/ directories.

kaweue commented on zuul/zuul-merger.service:
```diff
@@ -0,0 +1,13 @@
+Type=simple
+User=zuul
+Group=zuul
+#EnvironmentFile=-/etc/sysconfig/zuul
```
Is this required #EnvironmentFile=-/etc/sysconfig/zuul ?

jacek-izykowski replied:
No.

kaweue commented on zuul/zuul-server.service:
```diff
@@ -0,0 +1,13 @@
+Type=simple
+User=zuul
+Group=zuul
+#EnvironmentFile=-/etc/sysconfig/zuul
```
Is this required #EnvironmentFile=-/etc/sysconfig/zuul ?

jacek-izykowski replied:
No. Removed.

krdln committed:
add roles/zuul

krdln committed:
enable services in zuul role

add zuul user, group and directories

The zuul-server and zuul-merger services are now starting successfully.

sodar commented on zuul/zuul.conf:
```diff
@@ -0,0 +1,31 @@
+[gearman]
+server=10.84.12.188
```
`zuul/zuul.conf` can be templatized, so that when using Ansible to deploy/configure zuul Ansible can pass the correct IP address. I specifically talk about this http://docs.ansible.com/ansible/latest/template_module.html.

sodar commented on zuul/zuul.conf:
```diff
@@ -0,0 +1,31 @@
+state_dir=/var/lib/zuul
+git_dir=/var/lib/zuul/git
+url_pattern=
+status_url=10.84.12.188
```
Same comment regarding templates.

sodar commented on zuul/zuul.conf:
```diff
@@ -0,0 +1,31 @@
+url_pattern=
+status_url=10.84.12.188
+job_name_in_report=true
+zuul_url=http://10.84.12.188/merge-warrior
```
Same comment regarding templates.

sodar commented on zuul/zuul.conf:
```diff
@@ -0,0 +1,31 @@
+
+[merger]
+git_dir=/var/lib/zuul/git
+zuul_url=http://10.84.12.188/merge-warrior
```
Same comment regarding templates.

sodar commented on zuul/zuul.conf:
```diff
@@ -0,0 +1,31 @@
+log_config=/etc/zuul/gearman-logging.conf
+
+[gerrit]
+server=review2.opencontrail.org
```
`zuul/zuul.conf` can be templatized, so that when using Ansible to deploy/configure zuul Ansible can pass the correct Gerrit address. I specifically talk about this http://docs.ansible.com/ansible/latest/template_module.html.

sodar commented on zuul/zuul-server.service:
```diff
@@ -0,0 +1,13 @@
+Group=zuul
+#EnvironmentFile=-/etc/sysconfig/zuul
+ExecStart=/usr/local/bin/zuul-server -d
+ExecReload=/bin/kill -HUP $MAINPID
```
Does `zuul-merger` support `ExecReload`?

sodar commented on zuul/layout.yaml:
```diff
@@ -0,0 +1,87 @@
+    check:
+      - WinCI-Proxy
+
+  - name: Juniper/contrail-common
```
Will this list need to contain each project on Gerrit?

jacek-izykowski replied:
Yes.

Commit:
Use zuul specific apache2 config

Commit:
Moved apache2-zuul.conf

krdln committed:
update ansible to use apache2/conf-available and enabled

krdln committed:
copy vault-enrypted id_rsa to the zuul machine

The id_rsa file should be created using
    ansible-vault create id_rsa

Note: as for now, this file is not actually stored anywhere.

Commit:
Start handle only "recheck windows" comment triggers

krdln committed:
start zuul services automatically

Commit:
zuul.conf is a template for ansible

Commit:
Cleanup of zuul services cfg files: server, merger

Commit:
Renamed WinCI-Proxy to ci-contrail-windows

Commit:
Ansible for zuul - retrieving IP of zuul host

krdln commented on ansible/roles/zuul/tasks/main.yml:
```diff
@@ -49,6 +49,15 @@
+- name: 'generate zuul.conf from template'
+  template:
+     src: {{ config_dir }}/zuul.conf.j2
+     dest: /etc/zuul
```
Prefer using trailing slash when copying to a directory, ie: `/etc/zuul/`

krdln replied:
Actually, I'm not even sure if `dest` can be a directory when using a template, I'd go with a full path here.

Commit:
Ansible for zuul: syntax error fixed

Commit:
Zuul should not post any comments for now. Removed not needed projects

Commit:
Dest for template task in Zuul Ansible fixed

krdln committed:
put files for zuul in idiomatic location

Move them to roles/zuul/files and roles/zuul/templates

krdln committed:
add encrypted keys in zuul/vault

note: These are encrypted with a new vault key

krdln committed:
add review2 to known hosts

kaweue reviewed (changes requested):
I think we can add both review and review2 to known hosts

krdln committed:
add review to known hosts

krdln committed:
move deploykey from zuul vault to builder vault

krdln committed:
add zuul group to inventory

krdln committed:
anble apache2 cgi module in playbook

krdln committed:
Create ~zuul/.ssh directory

Now ansible runs correctly from a clean ubuntu server template

krdln committed:
detect when templated zuul.conf changed
